fvm/src/kernel/default.rs

```diff
@@ -745,18 +742,23 @@ impl<C> ActorOps for DefaultKernel<C>
 where
     C: CallManager,
 {
-    fn resolve_address(&self, address: &Address) -> Result<Option<ActorID>> {
-        self.call_manager.state_tree().lookup_id(address)
+    fn resolve_address(&self, address: &Address) -> Result<ActorID> {
```

Comment:
This is going to cause some problems, as actors typically treat kernel errors as fatal and dont look into them.

Also, semantically, is NotFound even a thing? Thats what Options are for.

Reply:
We check for _specific_ error codes in specific syscalls (when expected).